kod/object/active/holder/nomoveon/battler/monster/giaratk.kod

```diff
@@ -0,0 +1,131 @@
+% Meridian 59, Copyright 1994-2012 Andrew Kirmse and Chris Kirmse.
+% All rights reserved.
+%
+% This software is distributed under a license that is described in
+% the LICENSE file that accompanies it.
+%
+% Meridian is a registered trademark.
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+GiantRatKing is Monster
+
+constants:
+
+   include blakston.khd
+
+resources:
+
+   giantratkkoc_name_rsc = "konapyijoa"
+   giantratkname_rsc = "Rat King"
+   giantratkicon_rsc = ratking.bgf
+   giantratkdesc_rsc = \
+   "The giant rat king bares its yellow teeth in defiance.  A "
+   "smell of rot emanates from its filthy coat."
+
+   giantratkdead_icon_rsc = ratkingX.bgf
+   giantratkdead_name_rsc = "dead rat king"
+
+   giantratksound_hit = rat_atkh.wav
+   giantratksound_miss = rat_atkm.wav
+   giantratksound_death = rat_dth.wav
+   giantratksound_aware = rat_awr.wav
+
+classvars:
+
+   vrKocName = giantratkkoc_name_rsc
+   vrName = giantratkname_rsc
+   vrIcon = giantratkicon_rsc
+   vrDesc = giantratkdesc_rsc
+   vrDead_icon = giantratkdead_icon_rsc
+   vrDead_name = giantratkdead_name_rsc
+
+   viTreasure_type = TID_RAT_KING
+
+   viSpeed = SPEED_FAST
+   viAttack_type = ATCK_WEAP_CLAW
+   viAttributes = 0
+   viLevel = 60
+   viDifficulty = 5
+   viVisionDistance = 1
+   viKarma = 0
+   viDefault_behavior = AI_FIGHT_AGGRESSIVE | AI_FIGHT_SWITCHALOT
+   viCashmin = 150
+   viCashmax = 250
+
+   vrSound_hit = giantratksound_hit
+   vrSound_miss = giantratksound_miss
+   vrSound_aware = giantratksound_aware
+   vrSound_death = giantratksound_death
+
+properties:
+
+   piAnimation = ANIM_NONE
+
+messages:
+
+   Constructed()
+   {
+      plResistances = [ [-ATCK_SPELL_FIRE, -20],
+                        [ATCK_WEAP_BLUDGEON, -10],
+                        [ATCK_WEAP_SLASH, 20] ];
+                        
+      piHit_Points = 700;
+      piMax_Hit_Points = 700;                  
+                        
+      propagate;
+   }
+
+   MonsterAttack(what = $)
+   {
+      % add in any attack animation calls
+      Send(self,@DoSlash);
+
+      return;
+   }
+
+   DoSlash()
+   {
+      piAnimation = ANIM_ATTACK;
+      Send(poOwner,@SomethingChanged,#what=self);
+      piAnimation = ANIM_NONE;
+
+      return;
+   }
+
+   SendLookAnimation()
+   {
+      AddPacket(1,ANIMATE_CYCLE,4,125,2,2,2,6);
+
+      return;
+   }
+
+   SendMoveAnimation()
+   {
+      AddPacket(1,ANIMATE_CYCLE,4,75,2,2,2,6);
+
+      return;
+   }
+
+   SendAnimation()
+   {
+      if piAnimation = ANIM_ATTACK
+      {
+         AddPacket(1,ANIMATE_ONCE,4,100,2,7,2,11,2,1);
+
+         return;
+      }
+
+      % if no body animation
+
+      propagate;
+   }
+
+   IsAlly(target = $)
```

Comment:
Could have it see giant rats as allies, will make it more likely to defend them (this of course is dependent on what you want to do with the hatred list/brain).